frontend/lint/eslint-rules/no-unconditional-metabase-links-render.js

```diff
@@ -0,0 +1,217 @@
+
+    /**
+     * @typedef DefaultImport
+     * @property {true} isDefault
+     * @property {string} source
+     *
+     * @typedef NamedImport
+     * @property {string} named
+     * @property {string} source
+     *
+     * @param {object} node
+     * @param {DefaultImport|NamedImport} parameter
+     */
+    function getImportedModuleNode(node, { isDefault, named, source }) {
+      if (getImportNodeLocation(node) === source) {
+        const variables = context.getDeclaredVariables(node);
+        if (isDefault) {
+          return variables.find(
+            variable => variable.defs[0].node.type === "ImportDefaultSpecifier",
+          );
+        }
+
+        // Named import
+        return variables.find(
+          variable =>
+            variable.defs[0].node.type === "ImportSpecifier" &&
+            variable.name === named,
+        );
+      }
+
+      return undefined;
+    }
+
+    return {
+      ImportDeclaration(node) {
+        if (
+          getImportedModuleNode(node, {
+            isDefault: true,
+            source: "metabase/lib/settings",
+          })
+        ) {
+          metabaseSettings = getImportedModuleNode(node, {
+            isDefault: true,
+            source: "metabase/lib/settings",
+          });
+        }
+        if (
+          getImportedModuleNode(node, {
+            named: "getDocsUrl",
+            source: "metabase/selectors/settings",
+          })
+        ) {
+          isGetDocsUrlSelectorImported = true;
+        }
+        if (
+          getImportedModuleNode(node, {
+            named: "getLearnUrl",
+            source: "metabase/selectors/settings",
+          })
+        ) {
+          isGetLearnUrlSelectorImported = true;
+        }
+        if (
+          getImportedModuleNode(node, {
+            named: "getShowMetabaseLinks",
+            source: "metabase/selectors/whitelabel",
+          })
+        ) {
+          isGetShowMetabaseLinksSelectorImported = true;
+        }
+      },
+      CallExpression(node) {
+        // call `getDocsUrl` selector
+        if (
+          isGetDocsUrlSelectorImported &&
+          !isGetShowMetabaseLinksSelectorImported &&
+          node?.callee?.type === "Identifier" &&
+          node?.callee?.name === "getDocsUrl"
+        ) {
+          context.report({
+            node,
+            message: ERROR_MESSAGE,
+          });
+        }
+
+        // call `getLearnUrl` selector
+        if (
+          isGetLearnUrlSelectorImported &&
+          !isGetShowMetabaseLinksSelectorImported &&
+          node?.callee?.type === "Identifier" &&
+          node?.callee?.name === "getLearnUrl"
+        ) {
+          context.report({
+            node,
+            message: ERROR_MESSAGE,
+          });
+        }
+
+        // call `MetabaseSettings.learnUrl` or `MetabaseSettings.docsUrl`
+        if (
+          metabaseSettings?.references.some(
+            reference => reference.identifier === node?.callee?.object,
+          ) &&
```

Comment:
This is where we check that the default import from `metabase/lib/settings` is used to call `learnUrl()` or `docsUrl()`.

Previously I only checked if it's named `MetabaseSettings`, but the default import could be named anything, so it's better to check the reference instead. e.g. This is where we call it `Settings` instead of `MetabaseSettings`'
https://github.com/metabase/metabase/blob/e865ab9594ecd59fa4affb420fb3b9f25acaf4b8/frontend/src/metabase/admin/settings/slack/components/SlackStatus/SlackStatus.tsx#L45